util/template/kind.go

```diff
@@ -0,0 +1,20 @@
+package template
+
+import (
+	"strings"
+
+	jsonutil "github.com/argoproj/argo-workflows/v3/util/json"
+)
+
+type kind = string // defines the prefix symbol that determines the syntax of the tag
+
+var kinds []kind
+
+func parseTag(tag string) (kind, string) {
+	for _, k := range kinds {
+		if strings.HasPrefix(tag, k) {
+			return k, jsonutil.Fix(strings.TrimPrefix(tag, k))
+		}
+	}
+	return kindSimple, tag
```

Comment:
Not sure I'm a fan of the package wide variables each in different files. Makes reading and understanding the code difficult.

Reply:
I think you're correct, I'll put them into one place